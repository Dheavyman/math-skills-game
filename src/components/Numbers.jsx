import React from 'react';

const arrayOfNumbers = Array.from({ length: 9 }, (_, index) => index + 1);

/**
 * Numbers component
 *
 * @returns {object} React element
 */
const Numbers = () => (
  <div className="card text-center mt-5">
    <div className="number">
      {arrayOfNumbers.map((number, index) => (
        <span key={index.toString()}>{number}</span>
      ))}
    </div>
  </div>
);

export default Numbers;
