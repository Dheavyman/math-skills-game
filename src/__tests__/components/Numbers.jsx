import React from 'react';
import TestRenderer from 'react-test-renderer';

import Numbers from '../../components/Numbers';

describe('Numbers component', () => {
  let props = {
    selectedNumbers: [],
    usedNumbers: [],
    handleSelectNumber: jest.fn()
  };

  it('should render component successfully', () => {
    const renderer = TestRenderer.create(<Numbers {...props} />);

    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('should call handle submit function when the numbers are clicked', () => {
    const renderer = TestRenderer.create(<Numbers {...props} />);

    renderer.root.findAllByType('span')[1].props.onClick();
    renderer.root.findAllByType('span')[5].props.onKeyUp({ key: 'Enter' });
    expect(props.handleSelectNumber).toHaveBeenCalledWith(2);
    expect(props.handleSelectNumber).toHaveBeenLastCalledWith(6);
  });

  it('should add class name "selected" to selected numbers', () => {
    props = {
      ...props,
      selectedNumbers: [
        ...props.selectedNumbers, 6
      ]
    };
    const renderer = TestRenderer.create(<Numbers {...props} />);

    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('should add class name "used" to used numbers', () => {
    props = {
      ...props,
      usedNumbers: [
        ...props.usedNumbers, 9, 4
      ]
    };
    const renderer = TestRenderer.create(<Numbers {...props} />);

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
