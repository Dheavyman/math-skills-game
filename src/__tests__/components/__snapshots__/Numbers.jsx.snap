// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Numbers component should add class name "selected" to selected numbers 1`] = `
<div
  className="card border-primary text-center mt-5"
>
  <div
    className="number"
  >
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      1
    </span>
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      2
    </span>
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      3
    </span>
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      4
    </span>
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      5
    </span>
    <span
      className="selected"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      6
    </span>
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      7
    </span>
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      8
    </span>
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      9
    </span>
  </div>
</div>
`;

exports[`Numbers component should add class name "used" to used numbers 1`] = `
<div
  className="card border-primary text-center mt-5"
>
  <div
    className="number"
  >
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      1
    </span>
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      2
    </span>
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      3
    </span>
    <span
      className="used"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      4
    </span>
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      5
    </span>
    <span
      className="selected"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      6
    </span>
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      7
    </span>
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      8
    </span>
    <span
      className="used"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      9
    </span>
  </div>
</div>
`;

exports[`Numbers component should render component successfully 1`] = `
<div
  className="card border-primary text-center mt-5"
>
  <div
    className="number"
  >
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      1
    </span>
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      2
    </span>
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      3
    </span>
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      4
    </span>
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      5
    </span>
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      6
    </span>
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      7
    </span>
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      8
    </span>
    <span
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      9
    </span>
  </div>
</div>
`;
